lib/trino-collect/src/main/java/io/trino/collect/cache/NonEvictableCache.java

```diff
@@ -15,12 +15,26 @@
 
 import com.google.common.cache.Cache;
 
+import java.util.concurrent.ExecutionException;
+import java.util.function.Supplier;
+
 /**
  * A {@link com.google.common.cache.Cache} that does not support eviction.
  */
 public interface NonEvictableCache<K, V>
         extends Cache<K, V>
 {
+    default V getUnchecked(K key, Supplier<V> loader)
```

Comment:
In Guava, `getUnchecked` is available on a LoadingCache only (and only without a loader).

I would prefer to stay in line with Guava APIs, so that we don't need to undo these simplifications, eg when switching back to Guava (should they fix the issue), or refactoring the code.

Thus, while I introduced all the "EvictableCache" related class, I wouldn't yet want to embrace them as "an API".

Reply:
If, however, we wanted to depart from Guava APIs, i would prefer we have a different signature

```
V <T extends Exception> get(K key, ThrowingSupplier<T> loader) throws T;
```

this would erase the (logically useless) distinction between `get` and `getUnchecked`, i.e. the caller wouldn't need to make the choice.

Reply:
@findepi and I talked about this off line.  The problem with an API like this is exceptions can cross threads when there are concurrent calls to a single element.  The cache synchronizes and shares the result for all callers, which means the threads will shared the stacks from the one thread that actuallu executed the loader.